api/v1alpha4/index/index.go

```diff
@@ -39,16 +47,16 @@ const (
 // managers cache.
 func AddMachineNodeIndex(ctx context.Context, mgr ctrl.Manager) error {
```

Comment:
```suggestion
func ByMachineNode(ctx context.Context, mgr ctrl.Manager) error {
```

Reply:
How about `MachineByNodeName` so the function when used would read `index.MachineByNodeName` - might be more clear on what is being indexed. wdyt?